arch/arm/soc/nordic_nrf5/nrf52/power.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <soc_power.h>
+#include <nrf_power.h>
+
+#if defined(CONFIG_SYS_POWER_DEEP_SLEEP)
+/* System_OFF is deepest Power state available, On exiting from this
+ * state CPU including all peripherals reset
+ */
+static void _system_off(void)
+{
+	nrf_power_system_off();
+}
+#endif
+
+void _issue_low_power_command(void)
+{
+	/* Issue _WFE Enent*/
+	__SEV();
```

Comment:
Do you want to sleep unconditionally? Because this will do this, regardless of whether an interrupt was already pending. If you want to sleep only if an interrupt is *not* pending then you should do:

__WFE();
__SEV();
__WFE();

Reply:
Just to further clarify this. If you  use the SEV/WFE/WFE sequence then  you will go to sleep even if there was an interrupt executed between the entry to this function and the second WFE. Not sure if that is what we want, since in that case that interrupt may have done something that the kernel needs to react to. The other option I suggested, WFE/SEV/WFE, will only go to sleep if there is no interrupts pending (first WFE), and then the pend bit will be cleared unconditionally (SEV/WFE)

Reply:
good Idea @carlescufi, we will change to WFE/SEV/WFE.